tokio/src/sync/mpsc/bounded.rs

```diff
@@ -52,6 +52,8 @@ impl<T> fmt::Debug for Receiver<T> {
 /// Creates a bounded mpsc channel for communicating between asynchronous tasks,
 /// returning the sender/receiver halves.
 ///
+/// The provided `buffer` size must be at least 1.
+///
```

Comment:
I recommend adding "Panic" section and moving this line to that. like [`Option::unwrap`](https://doc.rust-lang.org/nightly/core/option/enum.Option.html#panics-1)

Reply:
I'm +1 on adding a `# Panics` section, but I do think it's worth also documenting it here. It might be a bit redundant, but I'd probably write something like
```rust
/// Creates a bounded mpsc channel ... blah blah blah...
///
/// The channel will buffer up to the provided number of messages. Once the buffer is
/// full, attempts to `send` new messages will yield until there is buffer capacity remaining.
/// The provided buffer size must be at least 1.
///
/// All data  ... blah blah blah ...
///
/// # Examples
/// ... blah blah blah ...
///
/// # Panics
/// - If the buffer capacity is 0.

Reply:
Great suggestions, thank you both!